spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3811,6 +3811,9 @@ def strip_trailing_spaces(self):
         Remove trailing whitespace on leaving a non-string line containing it.
         Return the number of removed spaces.
         """
+        if not self.hasFocus():

```

Comment:
Please enclose this in another `if`, like this:

    if not running_under_pytest():

to avoid the failure in our tests.